cmake/support_pllgpu.cmake

```diff
@@ -0,0 +1,94 @@
+###############################################################################################
+# This file provides macros to process CUDA, CUDAToolkit, cuQuantum and MPI external libraries.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+set(CRAY_MPICH OFF)
+
+# Macro to aid in finding CUDA lib
+macro(findCUDA external_libs)
+    find_package(CUDA REQUIRED)
+    find_package(CUDAToolkit REQUIRED)
+    if(CUDA_FOUND)
+        target_link_libraries(${external_libs} INTERFACE CUDA::cudart)
+    endif()
+endmacro()
+
+# Macro to aid in finding MPI lib
+macro(findMPI external_libs)
+    if(ENABLE_MPI)
+        find_package(MPI REQUIRED)
+        if(MPI_FOUND)
+            message(STATUS "MPI found.")
+        else()
+            message(FATAL_ERROR "MPI is NOT found.")
+        endif()
+
+        string(FIND "${CMAKE_SYSTEM}" "cray" subStrIdx)
+        if(NOT subStrIdx EQUAL -1)
+            if(DEFINED ENV{CRAY_MPICH_VERSION} AND NOT "$ENV{CRAY_MPICH_VERSION}" STREQUAL "none")
+                find_library(MPI_GTL
+                NAMES libmpi_gtl_cuda.so libmpi_gtl_cuda.so.0 libmpi_gtl_cuda.so.0.0.0
+                HINTS ENV CRAY_LD_LIBRARY_PATH
+                ENV LD_LIBRARY_PATH
+                )
+            set(CRAY_MPICH ON)
```

Comment:
Is there another way we can set a flag in here, rather than pre-declaring it in the outer-most scope?

Reply:
Actually, we don't need this flag. What I do now is to find if there is `gtl.so`. If yes, let's link against it. If not, there will be error messages if the system is cray linux.